contrib/src/main/scala/akka/stream/contrib/Valve.scala

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.contrib
+
+import akka.stream.stage.{GraphStageLogic, GraphStageWithMaterializedValue, InHandler, OutHandler}
+import akka.stream._
+
+import scala.concurrent.{Future, Promise}
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Pause/ Resume a Flow
+  */
+trait ValveSwitch {
+  /**
+    * Change the state of the valve
+    *
+    * @param mode expected mode to switch on
+    *
+    * @return
+    *         Future[true] if the mode change
+    *         Future[false] if the mode didn't change
+    */
+  def flip(mode: SwitchMode): Future[Boolean]
+}
+
+/**
+  * Valve permit to control a flow (pause/resume) by using a ValveSwitch.
+
+  * After calling [[ValveSwitch#flip(mode: SwitchMode)]] the running instance of the [[Graph]] of [[FlowShape]] that materialized to the
+  * [[Valve]] will change the state of the valve
+  *
+  * After calling [[ValveSwitch#flip(Close)]]: the valve state is changed to Close if a push is requested the current element is saved in a buffer and if a pull is trigger the called will be saved.
+  *
+  * After calling [[ValveSwitch#flip(Open)]]: the valve state is changed to Open if a buffered Element is detected it will be automatically be push and a pull will be trigger if one was requested
+  *
+  * @param mode : state of the valve at the startup of the flow (by default Open)
+  */
+class Valve[A](mode: SwitchMode = SwitchMode.Open) extends GraphStageWithMaterializedValue[FlowShape[A, A], ValveSwitch] {
```

Comment:
I think factory methods for creating the Scala DSL and Java DSL flows would be nice. Look at the other existing stages here in contrib for examples.